src/transformers/utils/generic.py

```diff
@@ -306,12 +306,10 @@ def __init_subclass__(cls) -> None:
         `static_graph=True` with modules that output `ModelOutput` subclasses.
         """
         if is_torch_available():
-            import torch.utils._pytree
-
-            torch.utils._pytree._register_pytree_node(
+            _torch_pytree._register_pytree_node(
                 cls,
-                torch.utils._pytree._dict_flatten,
-                lambda values, context: cls(**torch.utils._pytree._dict_unflatten(values, context)),
+                _model_output_flatten,
+                _model_output_unflatten,
```

Comment:
I am wondering if it is possible to move this to `__init__`, so we don't need to do it twice: one for subclasses and one for ModelOutput itself.

Reply:
We need to register the class object rather than the instance object to the pytree registry. The `__init__` method takes `self`, the instance, as the argument. We need to do this with `cls` (via a classmethod). There are two solutions:

1. Register the subclasses in `ModelOutput.__init_subclass__()` and register the `ModelOutput` class in the global scope. (This PR)
2. Implement a metaclass and register the classes (both `ModelOutput` and its subclasses) in `metaclass.__new__()`.

Reply:
Oh, your are right! Thanks!